projects/angular-calendar/src/modules/common/calendar-a11y.provider.ts

```diff
@@ -0,0 +1,154 @@
+ *   public openDayEventsLM({ date, locale, isDrSuess }: CustomA11yParams): string {
+ *     if (isDrSuess) {
+ *       return `
+ *         ${formatDate(date, 'EEEE MMMM d', locale)}
+ *          Today you are you! That is truer than true! There is no one alive
+ *          who is you-er than you!
+ *       `;
+ *     }
+ *   }
+ * }
+ *
+ * // in your component that uses the calendar
+ * providers: [{
+ *  provide: CalendarA11y,
+ *  useClass: CustomCalendarA11y
+ * }]
+ * ```
+ */
+@Injectable()
+export class CalendarA11y implements CalendarA11yInterface {
+  /**
+   * Aria label for the badges/date of a cell
+   * @example: `Saturday October 19 1 event`
+   */
+  public monthCell({ day, locale }: A11yParams): string {
+    if (day.badgeTotal > 0) {
+      return `
+        ${formatDate(day.date, 'EEEE MMMM d', locale)}
+         ${pluralize('event', day.badgeTotal, true)},
+         click to expand
+      `;
+    } else {
+      return `${formatDate(day.date, 'EEEE MMMM d', locale)}`;
+    }
+  }
+
+  /**
+   * Aria label for the open day events start landmark
+   * @example: `Saturday October 19 expanded view`
+   */
+  public openDayEventsLM({ date, locale }: A11yParams): string {
+    return `
+      Beginning of expanded view for ${formatDate(date, 'EEEE MMMM dd', locale)}
+    `;
+  }
+
+  /**
+   * Aria label for alert that a day in the month view was expanded
+   * @example: `Saturday October 19 expanded`
+   */
+  public openDayEventsAlert({ date, locale }: A11yParams): string {
+    return `${formatDate(date, 'EEEE MMMM dd', locale)} expanded`;
+  }
+
+  /**
+   * Descriptive aria label for an event
+   * @example: `Saturday October 19th, Scott's Pizza Party, from 11:00am to 5:00pm`
+   */
+  public eventDescription({ event, locale }: A11yParams): string {
+    if (event.allDay === true) {
+      return this.allDayEventDescription({ event, locale });
+    }
+
+    const aria = `
+      ${formatDate(event.start, 'EEEE MMMM dd', locale)},
+      ${event.title}, from ${formatDate(event.start, 'hh:mm a', locale)}
+    `;
+    if (event.end) {
+      return aria + ` to ${formatDate(event.end, 'hh:mm a', locale)}`;
+    }
+    return aria;
+  }
+
+  /**
+   * Descriptive aria label for an all day event
+   * @example:
+   * `Scott's Party, event spans multiple days: start time October 19 5:00pm, no stop time`
+   */
+  public allDayEventDescription({ event, locale }: A11yParams): string {
+    const aria = `
+      ${event.title}, event spans multiple days:
+      start time ${formatDate(event.start, 'MMMM dd hh:mm a', locale)}
+    `;
+    if (event.end) {
+      return (
+        aria + `, stop time ${formatDate(event.end, 'MMMM d hh:mm a', locale)}`
+      );
+    }
+    return aria + `, no stop time`;
+  }
+
+  /**
+   * Aria label for the calendar event actions icons
+   * @returns 'Edit' for fa-pencil icons, and 'Delete' for fa-times icons
+   */
+  public actionButtonLabel({ label }: A11yParams): string {
```

Comment:
Let's make this a bit more generic by adding a new field to the interface here: https://github.com/mattlewis92/calendar-utils/blob/master/src/calendar-utils.ts#L37-L42

Something like `a11yLabel?: string;`

Then this function can just return the `a11yLabel` value of the action.

Reply:
This is a different repository (calendar-utils). How would you recommend making this change?

Reply:
You can make a PR with the change there and I'll merge and release it so you can update it here

Reply:
Oh, I didn't even realize that it was _your_ library. Haha